### PR TITLE
Warn and offer to fix iRT database if target is a standard and also in the library

### PR DESCRIPTION
**Pull request by kaipot**

### CODE REVIEW

File: pwiz_tools/Skyline/Model/Irt/IrtDb.cs [hunk 1]
```diff
@@ -591,5 +591,23 @@ private static void EnsureDocumentXmlTable(ISession session)
...
+        public void RemoveLibraryPeptides(IEnumerable<Target> peptides)
+        {
+            using (var session = OpenWriteSession())
+            using (var transaction = session.BeginTransaction())
+            {
+                foreach (var pep in peptides)
+                {
+                    using (var cmd = session.Connection.CreateCommand())
+                    {
+                        cmd.CommandText = @"DELETE FROM IrtLibrary WHERE Standard = 0 AND PeptideModSeq = ?";
```
Comment (nickshulman) [inline pwiz_tools/Skyline/Model/Irt/IrtDb.cs:604]: If you want to remove all non-standard peptides which have the same sequence as a standard peptide you could do:
DELETE FROM IrtLibrary L1 WHERE L1.Standard = 0 and L1.PeptideModSeq IN (SELECT L2.PeptideModSeq FROM IrtLibrary L2 WHERE L2.Standard = 1)
Afterwards, if you wanted to make sure that there were no duplicates you could do:
DELETE FROM IrtLibrary L1 WHERE L1.Id <> (SELECT MIN(L2.Id) FROM IrtLibrary L2 WHERE L2.PeptideModeSeq = L1.PeptideModSeq)

Comment (kaipot) [reply pwiz_tools/Skyline/Model/Irt/IrtDb.cs]: You're right, that's a better way to do it.

**Review by nickshulman — COMMENTED** [review]
Since there isn't any user interface involved, it might be simpler if you move the checking for duplicates and deleting them to "IrtDbManager.LoadBackground".

That way, all of the database reading/writing is happening on a background thread and you don't need to worry about what might happen if the Document changes in between calls to "SetDocument".

**Comment by kaipot** [comment]
> Since there isn't any user interface involved, it might be simpler if you move the checking for duplicates and deleting them to "IrtDbManager.LoadBackground".
> 
> That way, all of the database reading/writing is happening on a background thread and you don't need to worry about what might happen if the Document changes in between calls to "SetDocument".

Thanks, those are all good suggestions. I made the changes.

**Review by nickshulman — APPROVED** [review]
Looks good to me.

**Pull request merged**